text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,678 @@
+  - "onerous" here is an intentionally subjective term. At a minimum, "onerous"
+    legal/licensing terms include but are *not* limited to: non-disclosure
+    requirements, non-compete requirements, contributor license agreements
+    (CLAs) or equivalent, "non-commercial"/"research-only"/etc terms,
+    requirements conditional on the employer or employment of any particular
+    Rust developers, revocable terms, any requirements that create liability
+    for the Rust project or its developers or users, or any requirements that
+    adversely affect the livelihood or prospects of the Rust project or its
+    developers or users.
+- Neither this policy nor any decisions made regarding targets shall create any
+  binding agreement or estoppel by any party. If any member of an approving
+  Rust team serves as one of the maintainers of a target, or has any legal or
+  employment requirement (explicit or implicit) that might affect their
+  decisions regarding a target, they must recuse themselves from any approval
+  process regarding the target's tier status, though they may otherwise
+  participate in discussions.
+  - This requirement does not prevent part or all of this policy from being
+    cited in an explicit contract or work agreement (e.g. to implement or
+    maintain support for a target). This requirement exists to ensure that a
+    developer or team responsible for reviewing and approving a target does not
+    face any legal threats or obligations that would prevent them from freely
+    exercising their judgment in such approval, even if such judgment involves
+    subjective matters or goes beyond the letter of these requirements.
+- Tier 3 targets should attempt to implement as much of the standard libraries
+  as possible and appropriate (`core` for most targets, `alloc` for targets
+  that can support dynamic memory allocation, `std` for targets with an
+  operating system), but may leave some code unimplemented (either unavailable
+  or stubbed out as appropriate), whether because the target makes it
+  impossible to implement or challenging to implement. The authors of pull
+  requests are not obligated to avoid calling any portions of the standard
+  library on the basis of a tier 3 target not implementing those portions.
+- The target must provide documentation for the Rust community explaining how
+  to build for the target, using cross-compilation if possible. If the target
+  supports running tests (even if they do not pass), the documentation must
+  explain how to run tests for the target, using emulation if possible or
+  dedicated hardware if necessary.
+- Tier 3 targets must not impose burden on the authors of pull requests, or
+  other developers in the community, to maintain the target. In particular,
+  do not post comments (automated or manual) on a PR that suggests a block on
+  the PR based on the target. (A PR author may choose to help with a tier 3
+  target, but is not required to.)
```

Comment:
I would like to leave this requirement *relatively* subjective, because there are corner cases. For instance, on GitHub, linking to a PR from another issue/PR will create a backlink in the linked PR, and I certainly don't think *that* should be considered excessive (as long as it isn't abused). And if (for instance) a tier 2 target was currently in the process of pushing to tier 1 by fixing all testsuite failures, I think it'd be reasonable to have a single comment on a PR currently in flight as part of coordination efforts.

That said, I'm going to strengthen the language a little to make it clearer and less ambiguous.

Reply:
I've now strengthened these requirements substantially, while still carving out a narrow allowance for backlinks.